### PR TITLE
Improved JSDocs and inline comments for React components

### LINKED ISSUE #374: Improve documentation to make code more accessible
From David Goldstein: 

"One overall comment is I would add comments to the top of each new file detailing what is does and what the parent component is (it was a little tricky to differentiate between some of the side components off the bat). I might also fill out the type of param instead of leaving it at {*}"

### PR DESCRIPTION
**Pull request by spmcbride1201**
I improved the JSDocs and inline comments for all React components in the app and fixed a few minor bugs that I noticed while going through the app. I branched off a weird state, so my commit history is messed up. It shouldn't be an issue, as I plan to squash-merge.

Resolves #374

### CODE REVIEW

File: src/components/DygraphContainer.js [hunk 1]
```diff
@@ -36,29 +49,46 @@ export default class DygraphContainer extends React.Component {
...
-    options.legend = "always";
-    options.axisLineColor = "rgb(200, 200, 200)";
-    options.gridLineColor = "rgb(200, 200, 200)";
-    options.colors = ["#0aab2a", "#002e6e"];
-    options.gridLinePattern = [1, 3];
-    options.animatedZooms = true;
-    options.drawAxesAtZero = true;
-    options.labelsSeparateLines = true;
-    options.includeZero = true;
+    // Default Dygraph Options
```
Comment (spmcbride1201) [inline src/components/DygraphContainer.js]: I noticed that I was always klobbering the options passed in as props, so I fixed to make sure the props take precedence over the defaults.

File: src/components/Button.js [hunk 2]
```diff
@@ -1,28 +1,32 @@
...
   tabIndex: PropTypes.number,
   type: PropTypes.oneOf(["danger", "info", "primary", "secondary", "warning"])
 };
 
+/**
+ * General purpose button
+ * @param {Object} props - see propTypes 
+ * @returns JSX.Element
+ */
 function Button({
```
Comment (dgoldstein1) [inline src/components/Button.js]: yay! comments are great!

File: src/components/DygraphContainer.js (hunk 1 above)
Comment (dgoldstein1) [reply src/components/DygraphContainer.js]: good idea

File: src/components/GMBasicMetrics.js [hunk 3]
```diff
@@ -11,6 +11,10 @@ GMBasicMetrics.propTypes = {
   title: PropTypes.string.isRequired
 };
 
+/**
+ * Basic coponent for rendering a group of related metrics. Includes sparklines
```
Comment (dgoldstein1) [inline src/components/GMBasicMetrics.js]: spelling

File: src/components/GMGrid.js [hunk 4]
```diff
@@ -138,9 +137,10 @@ class GMGrid extends Component {
...
   /**
    * Renders a dashboard as a responsive grid
-   * @param {*} dashboard 
+   * @param {Object} dashboard 
    */
   renderDashboard(dashboard) {
+    // While this parent div looks superfluous, it is needed to ensure the proper vertical heigh of the dashboard
     return (
       <div>
         <ResponsiveReactGridLayout
```
Comment (dgoldstein1) [inline src/components/GMGrid.js]: these comments help readability a lot

File: src/components/NavButton.js [hunk 5]
```diff
@@ -3,25 +3,28 @@ import { NavLink } from "react-router-dom";
...
+  to: PropTypes.string.isRequired, // route the button should navigate to
   type: PropTypes.oneOf(["danger", "info", "primary", "secondary", "warning"])
 };
-
+/** 
+ * General purpose button used for client-side navigation anywhere outside of sidebar
+ * @param {Object} props - refer to propTypes
+ * */
 function NavButton({
   icon,
```
Comment (dgoldstein1) [inline src/components/NavButton.js]: so awesome!

**Pull request merged**